corehq/apps/users/management/commands/fix_commcare_version.py

```diff
@@ -15,28 +20,52 @@ class Command(BaseCommand):
 
     def handle(self, *args, **kwargs):
         db = CommCareUser.get_db()
-        count = get_doc_count_by_type(db, 'CommCareUser')
-        all_ids = chain(iter_all_doc_ids(db, 'CommCareUser'))
-        iter_update(db, update_commcare_version, with_progress_bar(all_ids, count), verbose=True)
+        domain_names = Domain.get_all_names()
+
+        for domain in domain_names:
+            if get_migration_complete(domain, 'fix_commcare_version'):
+                continue
+            count = get_doc_count_by_domain_type(db, domain, 'CommCareUser')
+            all_ids = (row['id'] for row in paginate_view(
+                db,
+                'by_domain_doc_type_date/view',
+                chunk_size=10000,
+                startkey=[domain, 'CommCareUser'],
+                endkey=[domain, 'CommCareUser', {}],
+                include_docs=False,
+                reduce=False,
+            ))
+            set_migration_started(domain, 'fix_commcare_version')
+            try:
+                res = iter_update(db, update_commcare_version, with_progress_bar(all_ids, count), max_retries=0)
```

Comment:
Why `max_retires` is set to 0 here? There are high chances of getting a ResouceConflict while trying to save a batch, I think keeping the default should be fine.

Reply:
Setting this to 0 explicitly because @esoergel  suggested disabling retries. The reason is 
> if you have an error like some weird old prod data that doesn't match the format you're anticipating, then simply running it again won't change anything. but at any rate, I don't think it's worth preemptively guarding against that - just check to ensure the command succeeded, and if it didn't, you'll have an exception with a traceback explaining what the problem is

Now read his words again, I get confused about what does "that" in "preemptively guarding against that" refers to...  Maybe he means it doesn't worth disabling retry to guard against weird old prod data? Agree retry will help with `ResourceConflict`, changed it to default in 61add00b3be6f4a2da7964a0e8b0e8bdebb7da71

Reply:
Ahhhh I see. Thank you! 😭 At this point I wish there are no weird old prod data. If there is, I'll tackle them separately since I've print error_ids to the console.